connectors/rsk.go

```diff
@@ -151,183 +121,151 @@ func (rsk *RSK) GasPrice() (*big.Int, error) {
 }
 
 func (rsk *RSK) HashQuote(q *types.Quote) (string, error) {
-	var err error
-	results := new([]interface{})
 	opts := bind.CallOpts{}
+	var results [32]byte
 
 	pq, err := parseQuote(q)
 	if err != nil {
 		return "", err
 	}
 
 	for i := 0; i < retries; i++ {
-		err = rsk.lbc.Call(&opts, results, "hashQuote", pq)
-		if len(*results) > 0 {
+		results, err = rsk.lbc.HashQuote(&opts, pq)
+		if err == nil {
 			break
 		}
 		time.Sleep(sleepTime)
 	}
-	if len(*results) == 0 {
-		return "", fmt.Errorf("error calling hashQuote %v: %v", pq, err)
+	if err != nil {
+		return "", fmt.Errorf("error calling HashQuote: %v", err)
 	}
-	arr := *results
-	bts := getBytes(arr[0])
-
-	return hex.EncodeToString(bts), nil
+	return hex.EncodeToString(results[:]), nil
 }
 
 func (rsk *RSK) GetFedSize() (int, error) {
 	var err error
-	results := new([]interface{})
 	opts := bind.CallOpts{}
+	var results *big.Int
 
 	for i := 0; i < retries; i++ {
-		err = rsk.bridge.Call(&opts, results, "getFederationSize")
-		if len(*results) > 0 {
+		results, err = rsk.bridge.GetFederationSize(&opts)
+		if results != nil {
 			break
 		}
 		time.Sleep(sleepTime)
 	}
-	if len(*results) == 0 {
-		return 0, fmt.Errorf("error calling getFederationSize: %v", err)
-	}
-	out := *results
-	size := *abi.ConvertType(out[0], new(*big.Int)).(**big.Int)
-	sizeInt, err := strconv.Atoi(size.String())
 	if err != nil {
-		return 0, fmt.Errorf("error parsing value from getFederationSize: %v", err)
+		return 0, fmt.Errorf("error calling GetFederationSize: %v", err)
 	}
 
+	sizeInt, err := strconv.Atoi(results.String())
+	if err != nil {
+		return 0, fmt.Errorf("error converting federation size to int. error: %v", err)
+	}
 	return sizeInt, nil
 }
 
 func (rsk *RSK) GetFedThreshold() (int, error) {
 	var err error
-	results := new([]interface{})
 	opts := bind.CallOpts{}
+	var results *big.Int
 
 	for i := 0; i < retries; i++ {
-		err = rsk.bridge.Call(&opts, results, "getFederationThreshold")
-		if len(*results) > 0 {
+		results, err = rsk.bridge.GetFederationThreshold(&opts)
+		if results != nil {
 			break
 		}
 		time.Sleep(sleepTime)
 	}
-	if len(*results) == 0 {
-		return 0, fmt.Errorf("error calling getFederationThreshold: %v", err)
+	if err != nil {
+		return 0, fmt.Errorf("error calling GetFederationThreshold: %v", err)
 	}
-	out := *results
-	size := *abi.ConvertType(out[0], new(*big.Int)).(**big.Int)
-	sizeInt, err := strconv.Atoi(size.String())
+
+	sizeInt, err := strconv.Atoi(results.String())
 	if err != nil {
-		return 0, fmt.Errorf("error parsing value from getFederationSize: %v", err)
+		return 0, fmt.Errorf("error converting federation size to int. error: %v", err)
 	}
 
 	return sizeInt, nil
 }
 
 func (rsk *RSK) GetFedPublicKeyOfType(index int) (string, error) {
```

Comment:
Shouldn't be named OfType if it does not take the type

Reply:
done.